Framework/DataObjects/src/TimeSplitter.cpp

```diff
@@ -37,6 +37,39 @@ TimeSplitter::TimeSplitter(const DateAndTime &start, const DateAndTime &stop) {
   clearAndReplace(start, stop, DEFAULT_VALUE);
 }
 
+TimeSplitter::TimeSplitter(const TableWorkspace_sptr &tws, const DateAndTime &offset) {
```

Comment:
I think this constructor should accept `const ITableWorkspace_sptr` instead of `const ITableWorkspace_sptr` so that we can `dynamic_pointer_cast` to `SplitterWorkspace` first and then to `TableWorkspace`, and instantiate accordingly by invoking corresponding private methods.

Reply:
As we all have agreed, there will be two separate constructors in this case.